File: Backend Code/src/main/java/com/LabResourceUtilizationPlatform/Service/DashBoardService.java
package com.LabResourceUtilizationPlatform.Service;

import com.LabResourceUtilizationPlatform.Dtos.Response.*;
import com.LabResourceUtilizationPlatform.Entity.Institution;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.Query;

import java.util.List;

public interface DashBoardService {

    WeeklyUtilizationResponse getWeeklyUtilization();
    TechnicianDashboardResponse getTechnicianDashboard();
    StudentDashboardResponse getStudentDashboard();

    ResearcherDashboardResponse getResearcherDashboard();

    FacultyDashboardResponse getFacultyDashboard();

    LabManagerDashboardResponse getLabManagerDashboard();

    DepartmentHeadDashboardResponse getDepartmentHeadDashboard();

    InstitutionAdminDashboardResponse getInstitutionAdminDashboard();

    SystemAdminDashboardResponse getSystemAdminDashboard();
}

File: Frontend Code/src/services/dashboardService.js
import api from "./api";

const dashboardService = {
  getWeeklyUtilization: async () => {
    const response = await api.get("/dashboard/weekly-utilization");
    return response.data;
  },

  getStudentDashboard: async () => {
    const response = await api.get("/dashboard/student");
    return response.data;
  },

  getResearcherDashboard: async () => {
    const response = await api.get("/dashboard/researcher");
    return response.data;
  },

  getFacultyDashboard: async () => {
    const response = await api.get("/dashboard/faculty");
    return response.data;
  },

  getTechnicianDashboard: async () => {
    const response = await api.get("/dashboard/technician");
    return response.data;
  },

  getLabManagerDashboard: async () => {
    const response = await api.get("/dashboard/lab-manager");
    return response.data;
  },

  getDepartmentHeadDashboard: async () => {
    const response = await api.get("/dashboard/department-head");
    return response.data;
  },

  getInstitutionAdminDashboard: async () => {
    const response = await api.get("/dashboard/institution-admin");
    return response.data;
  },

  getSystemAdminDashboard: async () => {
    const response = await api.get("/dashboard/system-admin");
    return response.data;
  },
};

export default dashboardService;